homeassistant/components/wemo.py

```diff
@@ -36,11 +37,37 @@
 
 _LOGGER = logging.getLogger(__name__)
 
+
+def coerce_host_port(value):
+    """Validate that provided value is either just host or host:port.
+
+    Returns (host, None) or (host, port) respectively.
+    """
+    host, _, port = value.partition(':')
+
+    if not host:
+        raise vol.Invalid('host cannot be empty')
+
+    if port:
+        try:
+            port = cv.port(port)
+        except vol.Invalid as exc:
+            _slice = slice(value.index(':') + 1, len(value))
```

Comment:
I'm not sure I think this is more readable for users. I don't think all users know what `slice(12, 18, None)` means.

Reply:
Keeping default exception then? It is still not super descriptive, but my hope it is easy to spot errors regardless and this otherwise complicates things needlessly. *OR* replace this with dict config for consistency sake with other similar schemas
https://github.com/home-assistant/home-assistant/blob/9cfbd067d3fd9287a13fedbf3f77f24d8ac92d4b/homeassistant/components/android_ip_webcam.py#L134-L138

`vol.Invalid.__str__()` generates that through `repr(slice_instance)` so changing voluptuous may be better approach. Can't subclass slice and path being builtin list does specifically typecheck on slice :/

Reply:
I think the default exception is good enough.

Having a dict with separate host and port would be best but since that would be a breaking change, I don't think we should do that.

Reply:
Removed 'path' modification. I wish that worked nicer.